### PR TITLE
change perms for prettytable PKG-INFO to 644 for swift

### PR DESCRIPTION
**Pull request by CullenTaylor**
Swiftops user is unable to read prettytable package info, which prevents swift ring setup from completing. This file permission change should fix that.

### CODE REVIEW

File: roles/swift-ring/tasks/main.yml [hunk 1]
```diff
@@ -9,6 +9,9 @@
         dest=/etc/swift/ring_definition.yml mode=644 backup=yes
   when: not should_boostrap|success
 
+- name: change prettytable permissions
+  shell: chmod 644 /usr/local/lib/python2.7/dist-packages/prettytable-0.7.2.egg-info/PKG-INFO
```
Comment (pwplusnick) [inline roles/swift-ring/tasks/main.yml]: Use `file` Ansible module http://docs.ansible.com/ansible/file_module.html

**Comment by CullenTaylor** [comment]
@pwplusnick fixed. thanks

**Comment by paulczar** [comment]
Can you provide details on what is wrong here?  we don't hit this problem.

**Comment by CullenTaylor** [comment]
@paulczar when deploying swift using master, I've hit the following error in all of our softlayer-based environments.

```
TASK: [swift-ring | setup swift rings] ****************************************
Wednesday 16 December 2015  12:27:51 -0600 (0:00:00.267)       0:42:04.985 ****
failed: [swiftnode1] => {"changed": true, "cmd": ["sudo", "-u", "swiftops", "/usr/local/bin/swifttool", "-i", "/home/swiftops/.ssh/id_rsa", "bootstrap", "--config", "/etc/swift/ring_definition.yml", "--outdir", "/etc/swift"], "delta": "0:00:00.091544", "end": "2015-12-16 18:27:51.714120", "rc": 1, "start": "2015-12-16 18:27:51.622576", "warnings": []}
stderr: Traceback (most recent call last):
  File "/usr/local/bin/swifttool", line 5, in <module>
    from pkg_resources import load_entry_point
  File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 3138, in <module>
    @_call_aside
  File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 3124, in _call_aside
    f(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 3151, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 652, in _build_master
    ws = cls()
  File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 645, in __init__
    self.add_entry(entry)
  File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 701, in add_entry
    for dist in find_distributions(entry, True):
  File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2139, in find_on_path
    path_item, entry, metadata, precedence=DEVELOP_DIST
  File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2521, in from_location
    py_version=py_version, platform=platform, **kw
  File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2835, in _reload_version
    md_version = _version_from_file(self._get_metadata(self.PKG_INFO))
  File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2486, in _version_from_file
    line = next(iter(version_lines), '')
  File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2654, in _get_metadata
    for line in self.get_metadata_lines(name):
  File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 1643, in get_metadata_lines
    return yield_lines(self.get_metadata(name))
  File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 1635, in get_metadata
    return self._get(self._fn(self.egg_info, name))
  File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 1746, in _get
    with open(path, 'rb') as stream:
IOError: [Errno 13] Permission denied: '/usr/local/lib/python2.7/dist-packages/prettytable-0.7.2.egg-info/PKG-INFO'

FATAL: all hosts have already failed -- aborting
```

**Comment by j2sol** [comment]
How was your prettytable installed? In our CI it appears to come from a Ubuntu package, `python-prettytable` which has the correct permissions for users to access it. 

From your traceback, it seems like your prettytable was installed via pip at some point.

**Comment by j2sol** [comment]
Actually, I'm not even sure how prettytable is getting involved here. I've been able to get past the point in which your traceback happens with and without prettytable installed. I don't think swifttool actually needs it, but more to the point, a fresh pip install of prettytable shows:

```
 ls -al /usr/local/lib/python2.7/dist-packages/prettytable-0.7.2-py2.7.egg-info
total 36
drwxr-sr-x   2 root staff  4096 Dec 17 11:27 .
drwxrwsr-x 149 root staff 12288 Dec 17 11:27 ..
-rw-r--r--   1 root staff     1 Dec 17 11:27 dependency_links.txt
-rw-r--r--   1 root staff    96 Dec 17 11:27 installed-files.txt
-rw-r--r--   1 root staff   710 Dec 17 11:27 PKG-INFO
```

so the permissions are fine coming from pypi. I really question how your install of prettytable (which isn't required) got messed up.

Closing this as the change is inappropriate. Conversation can continue though.

**Pull request closed without merging**